### PR TITLE
Added basic timeline visualizer to dashboard

### PR DESCRIPTION
**Pull request by am-zhang**
# Description

This PR adds a basic timeline that displays data from perceive events from loco_mc_agent.py in real time. It includes

- Scrolling, zooming, and tooltips for the timeline
- Adding package dependencies for the visjs Timeline framework
- Formatting to display JSON in the tooltips (hover event)
- Removing the timeline handshake text and other small visual changes
- Temporarily deregistering the db_write hook due to volume of queries
- Updated gitignore for timeline log files

Run by entering `python craftassist_agent.py --log_timeline --enable_timeline`.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Screenshot:

Screen Shot 2021-06-29 at 11 19 26 AM

# Testing

Tested by entering commands from the dashboard and checking the timeline for changes.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: agents/loco_mc_agent.py [hunk 1]
```diff
@@ -330,8 +323,8 @@ def perceive(self, force=False):
             # Send data to the dashboard timeline
             hook_data = {
                 "name" : "perceive",
+                "datetime" : datetime.datetime.now(),
```
Comment (snyxan) [inline agents/loco_mc_agent.py:326]: We should record both start and end time of the event and show it in timeline.
This is also supported/reflected in visjs --> event items will have different length based on duration: https://visjs.github.io/vis-timeline/examples/timeline/groups/visibleGroups.html

File: droidlet/dashboard/web/src/components/Timeline/Timeline.js [hunk 2]
```diff
@@ -1,37 +1,76 @@
 /*
 Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (snyxan) [inline droidlet/dashboard/web/src/components/Timeline/Timeline.js:2]: Could you add some high-level description of this component here?

File: droidlet/memory/sql_memory.py [hunk 3]
```diff
@@ -1037,7 +1038,7 @@ def db_write(self, query: str, *args) -> int:
         query_dict = format_query(query, *args)
         hook_data = {
             "name" : "db_write", 
-            "time" : self.get_time(), 
+            "datetime" : datetime.datetime.now(),
```
Comment (snyxan) [inline droidlet/memory/sql_memory.py:1041]: same as above^

**Comment by snyxan** [comment]
In hook data, please also send agent's internal time, see: https://github.com/facebookresearch/droidlet/blob/main/agents/loco_mc_agent.py#L290

**Comment by snyxan** [comment]
For the following PRs, please also include necessary commands to test the feature. e.g. in order to run and see the timeline events, we should enable timeline option by running `python craftassist_agent.py --enable_timeline`. Someone could have missed this option and got confused by the empty timeline.

File: agents/loco_mc_agent.py (hunk 1 above)
Comment (am-zhang) [reply agents/loco_mc_agent.py:326]: One question I have about this is that the project plan states "the time is discrete, which means you can only select a time point when an activity occurred." This is also shown in the wireframes. I can add it right now but just to be clear, the project plan specs have changed?

Comment (snyxan) [reply agents/loco_mc_agent.py]: Our conclusion from offiline discussion: the timeline can either be in physical time mode where events have duration (start time & end time corresponded to the physical time) or in logical time mode where time is discrete and events occurs at discrete time point.
For now we are using physical time mode, but it's a good-to-have thing to be able to switch time mode -- putting it to stretch goal.

File: agents/loco_mc_agent.py [hunk 4]
```diff
@@ -328,10 +322,13 @@ def perceive(self, force=False):
...
             self.memory.tag(subj_memid=chat_memid, tag_text="unprocessed")
             # Send data to the dashboard timeline
+            end_time = datetime.datetime.now()
             hook_data = {
                 "name" : "perceive",
+                "start_datetime" : start_time,
+                "end_datetime" : end_time,
                 "speaker" : speaker, 
-                "time" : self.last_chat_time,
+                "agent_time" : self.last_chat_time,
```
Comment (snyxan) [inline agents/loco_mc_agent.py:331]: it's not the logical time the agent is using (line#314). You should use `self.get_time()` here.

Comment (am-zhang) [reply agents/loco_mc_agent.py]: Thanks for catching this! I've fixed it in [232a38a](https://github.com/facebookresearch/droidlet/pull/526/commits/232a38a5cf2fec5c57535a1f46187c785ed97dd4).

**Comment by kavyasrinet** [comment]
Changes look good to me after the fixes and @snyxan review.

**Review by snyxan — APPROVED** [review]
LGTM. Ship it!

**Pull request merged**